src/analyses/goto_rw.h

```diff
@@ -60,12 +68,27 @@ inline range_spect to_range_spect(const mp_integer &size)
 }
 
 // each element x represents a range of bits [x.first, x.second.first)
-class range_domaint:
-  public range_domain_baset,
-  public std::list<std::pair<range_spect, range_spect> >
+class range_domaint:public range_domain_baset
 {
+  using sub_typet=std::list<std::pair<range_spect, range_spect>>;
+  sub_typet data_;
+
 public:
-  virtual void output(const namespacet &ns, std::ostream &out) const;
+  void output(const namespacet &ns, std::ostream &out) const override;
+
+  using iterator=sub_typet::iterator;
+  using const_iterator=sub_typet::const_iterator;
```

Comment:
Is there any added value in this versatility? I'd go as far as saying: we use `typedef` everywhere else, so let's use `typedef` here as well.